### PR TITLE
extract eigenDecomposition from math.Linear and using strategy pattern

### PR DESCRIPTION
**Pull request by Dong-Ki-Lee**
extract eigenDecomposition from math.Linear and using strategy pattern
because the algorithm for decomposition can be changed in the future.

### CODE REVIEW

File: src/rapaio/math/linear/Linear.java [hunk 1]
```diff
@@ -49,73 +52,33 @@ public static RM chol2inv(RM R, RM B) {
...
         // Solve L'*X = Y;
-        for (int k = ref.getRowCount() - 1; k >= 0; k--) {
-            for (int j = 0; j < X.getColCount(); j++) {
-                for (int i = k + 1; i < ref.getRowCount(); i++) {
-                    X.increment(k, j, -X.get(i, j) * ref.get(i, k));
-                }
-                X.set(k, j, X.get(k, j) / ref.get(k, k));
-            }
-        }
+        multiply = new BackwordMultiply();
```
Comment (padreati) [inline src/rapaio/math/linear/Linear.java]: BackwardMultiply

**Comment by padreati** [comment]
I agree in general with some naming conventions:

- Backword is not correct, it should be Backward
- I do not like BackwardMultiply or ForwardMultiply. The standard names are BackwardSubstitution and ForwardSubstitution and I would like to keep the standard naming.

**Comment by Dong-Ki-Lee** [comment]
Ok, I resend this pull request after revise class name you requested.
Thank you for your comment.

**Comment by Dong-Ki-Lee** [comment]
I resend after revise

**Pull request closed without merging**